### ISSUE TITLE
Help with broken game

### ISSUE THREAD

**Issue by gmcgmc** [entropia/tip-toi-reveng#254]
Hi,

actually this is not an issue with the tttool but rather a cry for help ;)

Situation: I've a tiptoi with recorder (German). I (actually my children) recently got two games in Italian and I used tttool to change the language. And it worked flawless.

But one of the two gets stucked in a loop after starting a game (all of 3 available games). After selecting the game one should point the reader to the players' card to register the players to the game. But whatever the reader scans, it just repeats the same audio (asking to register the players). 

After reading the comments on Amazon Italy seems to be that other people have the same problem (I suppose with Italian Titpoi?). So this is probably some bug in Tiptoi's firmware and/or in the gme itself. Somebody seems to have fixed the problem by upgrading the firmware. As far as I understood there's currently no new firmware available for the tiptoi recorder ("mit Aufnahme Funktion") and as well there's no official way to change the firmware's language to Italian.

Details:
ID-Product 63
Name "Il mago della lettura"
Link to the gme file: https://ssl-static.ravensburger.de/db/applications/Il%20Mago%20della%20Lettura.gme

How to reproduce the problem:
2491 (choose game)
2802 (should be one of the player's card, supposedly should speak the color of this card) 
actually whatever code is scanned from the reader it just repeats the same audio.

Interestingly there's the German version of the game with the same codes.
ID-Product 7
gme: https://cdn.ravensburger.de/db/applications/Magors%20Lesezauber.gme

After roundtrip by changing only the product-id (I don't know how change it directly in the binary without raoundtrip), I've the same problem (well now in German :) ) On Amazon-DE there seems to be no comment about non playable games.

Tiptoi ID: HP135454 13070 Te4

I tried with another Tiptoi (as well "Te4") and I've the same problem.

Any idea what to do?

Bye
Giuseppe

**Comment by nomeata** [comment]
How exactly did you change the language? Using `tttool export` and `tttool assemble`, or with `tttool set-language`?

**Comment by gmcgmc** [comment]
Forgot to write it ;) of course with `tttool set-language`.

**Comment by nomeata** [comment]
Sorry, no helpful ideas then :-/

**Comment by uli42** [comment]
Maybe the game binary of the German version can be implanted into the
Italian version. This would require some addition to tttool by Joachim, I
guess.

Uli

Joachim Breitner ***@***.***> schrieb am So., 29. Aug. 2021,
12:06:

**Comment by nomeata** [comment]
`tttool set-language` leaves binaries in place. It only edits a few bytes in the header. So that can't really be it.

**Comment by uli42** [comment]
Yes, it can: if the Italian gme file is broken it will still be broken
after changing the language to German. So taking the German game blob
and replacing the blob in the Italian gme could be worth a try.
However, AFAIK tttool is missing a command to do something like this:
1. extract blob to file
2. replace blob from file.

**Comment by domenicov** [comment]
> Hi,
> 
> actually this is not an issue with the tttool but rather a cry for help ;)
> 
> Situation: I've a tiptoi with recorder (German). I (actually my children) recently got two games in Italian and I used tttool to change the language. And it worked flawless.
> 
> But one of the two gets stucked in a loop after starting a game (all of 3 available games). After selecting the game one should point the reader to the players' card to register the players to the game. But whatever the reader scans, it just repeats the same audio (asking to register the players).
> 
> After reading the comments on Amazon Italy seems to be that other people have the same problem (I suppose with Italian Titpoi?). So this is probably some bug in Tiptoi's firmware and/or in the gme itself. Somebody seems to have fixed the problem by upgrading the firmware. As far as I understood there's currently no new firmware available for the tiptoi recorder ("mit Aufnahme Funktion") and as well there's no official way to change the firmware's language to Italian.
> 
> Details: ID-Product 63 Name "Il mago della lettura" Link to the gme file: https://ssl-static.ravensburger.de/db/applications/Il%20Mago%20della%20Lettura.gme
> 
> How to reproduce the problem: 2491 (choose game) 2802 (should be one of the player's card, supposedly should speak the color of this card) actually whatever code is scanned from the reader it just repeats the same audio.
> 
> Interestingly there's the German version of the game with the same codes. ID-Product 7 gme: https://cdn.ravensburger.de/db/applications/Magors%20Lesezauber.gme
> 
> After roundtrip by changing only the product-id (I don't know how change it directly in the binary without raoundtrip), I've the same problem (well now in German :) ) On Amazon-DE there seems to be no comment about non playable games.
> 
> Tiptoi ID: HP135454 13070 Te4
> 
> I tried with another Tiptoi (as well "Te4") and I've the same problem.
> 
> Any idea what to do?
> 
> Bye Giuseppe

Hi Giuseppe, some friends gave the same game to my daughter a few days ago. I am in exactly the same situation as you. I know it's been a long time and maybe you've found a solution... impossible to talk to Tip-Toi technical support and I'm about to 'throw in the towel' :-) Thanks anyway, Domenico

**Comment by gmcgmc** [comment]
Hi (or ciao? :) ) @domenicov.

Unfortunately I didn't find any solution (and didn't try anything anymore...)

Best
Giuseppe
